ext/background/filter/jira.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+(() => {
+  registerFilter({
+    url_regex: [/atlassian\.net\/browse\//],
+    provider: 'jira',
+    provider_name: (_url) => 'Jira',
+    type: (_url) => 'ISSUE',
+    processors: [
+      {
+        source_id_processor: (url) => {
+          const hostname = toHostname(url);
+          const issueKey = toIssueKey(url);
+          if (!issueKey) {
+            return null;
+          }
+
+          return `${hostname}/${issueKey}`;
+        },
+        title_processor: (title) => {
+          if (title.endsWith(' - Jira')) {
+            title = trimLastPart(title, ' - ');
+          }
+          if (title.startsWith('[')) {
+            // The leading issue key will be included in the issue_info.
+            // Remove the leading leading "[AB-1] ".
+            title = title.replace(/^\[[^\]]+\]\s*/, '');
+          }
+          return title;
+        },
+        issue_info: (url) => {
+          const issueKey = toIssueKey(url);
+          if (issueKey) {
+            return {
+              issue_id: issueKey,
+            };
+          }
+
+          return null;
+        },
+      },
+    ],
+  });
+
+  function toIssueKey(url) {
+    const a = document.createElement('a');
```

Comment:
Never seen this before! Seems like it might be heavier than using `new URL()`?

Reply:
Agreed! Done https://github.com/range-labs/range-sync/pull/24/commits/186be2ccf3d95094b384cd52ed5cc4513fe78e39.